catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/QueryResponseImpl.java

```diff
@@ -302,7 +302,7 @@ private Result handleTake() {
         Result result = null;
         try {
             result = queue.take();
-            if (result == poisonPillResult) {
+            if (result == POISON_PILL_RESULT) {
```

Comment:
I think there are a couple of things we could do. We could override equals and then have relevance and distance set to always return `Double.MIN_VALUE` for the poison pill. Then in the equals method we could just check for both of those being set to that value. If, for some reason, that isn't feasible (i.e. some real result could come in with those values potentially), then we could also create a "poison pill metacard" inner class and set the metacard of the result to that and check for a specific title or field on the metacard that we know only exists on the poison pill.